raft/src/main/java/org/apache/kafka/raft/RaftClient.java

```diff
@@ -47,6 +47,14 @@
      */
     CompletableFuture<OffsetAndEpoch> append(Records records);
 
+    /**
+     * Append a control record to the log. The client must be in the leader state to
+     * accept a control record append. The control record append happens immediately.
+     *
+     * @param controlRecord the control record to be appended
+     */
+    OffsetAndEpoch appendControlRecord(Records controlRecord);
```

Comment:
I don't think we need this in the interface. We should only include the methods that would be needed by a caller. Control records should only be added by the implementation.